### PR TITLE
Key management docs overhaul

### PR DESCRIPTION
**Pull request by asaj**
## Description

_A few sentences describing the overall effects and goals of the pull request's commits.
What is the current behavior, and what is the updated/expected behavior with this PR?_

## Other changes

_Describe any minor or "drive-by" changes here._

## Tested

_An explanation of how the changes were tested or an explanation as to why they don't need to be._

## Related issues

- Fixes #[issue number here]

## Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

## Commit Message
_If using automerge, specify desired commit title_

_And desired commit message body_

### CODE REVIEW

File: packages/docs/operations-manual/key-management.md [hunk 1]
```diff
@@ -4,65 +4,209 @@
...
 
-The Account key is the most sensitive key because it can authorize the other keys and access locked gold. As it should be used quite infrequently, it is highly recommended for important Account keys (i.e. Validators or accounts with high balances) to remain as secure as possible. We recommend them to be offline, ideally in cold storage, or on a hardware wallet.
+Note that the validator signing key must first be used to generate a "proof-of-possession" indicating that key's willingness to be authorized as a signer on behalf of the Locked Gold Account.
 
-Signing keys are less sensitive, and more frequently used, than the Account key, but should still be safe-guarded. Validator signing keys, in particular, could be used to execute a double-signing attack if compromised, which may result in [slashing](../celo-codebase/protocol/proof-of-stake/penalties).
+
+### Authorizing a Validator Signer
+An undesignated key may be authorized as a validator signer on behalf of a Locked Gold Account by runnign the following commands:
```
Comment (kobigurk) [inline packages/docs/operations-manual/key-management.md]: runnign -> running

File: packages/docs/operations-manual/key-management.md [hunk 2]
```diff
@@ -4,65 +4,209 @@
...
+| ----------------------------------- |-----------------------------------------------------------------------------------|-------------------|
+| Undesignate                         | Default key type, cannot participate in proof-of-stake                            | Yes               |
+| Locked Gold Account (LGA)           | Used to lock and unlock cGLD and authorize other keys                             | Yes               |
+| Authorized vote signer              | Can vote on behalf of an LGA                                                      | Yes               |
+| Authorized validator (group) signer | Can register and manage a validator group on behalf of an LGA                     | No                |
+| Authorized validator signer         | Can register, manage a validator, and sign consensus messages on behalf of an LGA | No                |
+| Authorized validator BLS signer     | Used to sign blocks as a validator                                                | No                |
+| Authorized attestations signer      | Can sign attestations messages on behalf of a Locked Gold account                 | No                |
```
Comment (nambrot) [inline packages/docs/operations-manual/key-management.md]: I think its interesting to note that this technically is compatible, of course it would require one to touch the ledger with the spender app which I assume at some point there will be an option to do this?

File: packages/docs/operations-manual/key-management.md [hunk 3]
```diff
@@ -4,65 +4,209 @@
...
+A Locked Gold Account may have at most one authorized signer key of each type at any time. Once a signer key is authorized, the only way to deauthorize that key is to authorize a new key that has never previously been used as an authorized signer key for any Locked Gold Account. It follows then that a newly deauthorized key cannot be reauthorized for any account.
 {% endhint %}
 
-## Key Security
+## Undesignated
+
+By default, any private key generated in the Celo protocol does not have a specific role in proof-of-stake. These keys can be used to sign and send transactions that are unrelated to Celo's proof-of-stake protocol, including but by no means limited to transferring Celo Gold, transferring Celo Dollars, exchanging Celo Gold for Celo Dollars, etc.
+
+Undesignated keys can be designated as Locked Gold Accounts or authorized as signer keys on behalf of a Locked Gold Account by sending special transactions using `celocli`.
```
Comment (nambrot) [inline packages/docs/operations-manual/key-management.md]: I know  this is mostly written with respect to PoS, but the Account on `Accounts` isn't technically tied to LockedGold? I.e. for mobile wallets, you can create accounts to that one can set encryption keys. Smart contract developers could set an account to upload metadata to indicate trustworthiness.

File: packages/docs/operations-manual/key-management.md [hunk 4]
```diff
@@ -4,65 +4,209 @@
...
+
+Note that the vote signing key must first be used to generate a "proof-of-possession" indicating that key's willingness to be authorized as a signer on behalf of the Locked Gold Account.
+
+
+### Authorizing a Vote Signer
+An undesignated key may be authorized as a vote signer on behalf of a Locked Gold Account by runnign the following commands:
+
+```shell
+# Create a proof-of-possession. Note that the signer private key must be available.
+celocli account:proof-of-possession --account $LOCKED_GOLD_ACCOUNT --signer $KEY_TO_AUTHORIZE --useLedger
```
Comment (nambrot) [inline packages/docs/operations-manual/key-management.md]: Should we actually specify `--useLedger` on every command? It seems somewhat orthogonal

File: packages/docs/operations-manual/key-management.md [hunk 5]
```diff
@@ -4,65 +4,209 @@
...
 
-Signing keys are less sensitive, and more frequently used, than the Account key, but should still be safe-guarded. Validator signing keys, in particular, could be used to execute a double-signing attack if compromised, which may result in [slashing](../celo-codebase/protocol/proof-of-stake/penalties).
+
+### Authorizing a Validator Signer
+An undesignated key may be authorized as a validator signer on behalf of a Locked Gold Account by runnign the following commands:
+
+```shell
+# Create a proof-of-possession. Note that the signer private key must be available.
+# Note that the signing key should not be kept on a Ledger if it will be used to run a Validator
+celocli account:proof-of-possession --account $LOCKED_GOLD_ACCOUNT --signer $KEY_TO_AUTHORIZE --useLedger
```
Comment (nambrot) [inline packages/docs/operations-manual/key-management.md]: Isn't this `useLedger` in contradiction with the comment above it?

File: packages/docs/operations-manual/key-management.md (hunk 2 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management.md]: Yeah you can compile a "NO CONSENT" version and sideload that but it's not really an easy experience. If we want to have an attestations specific app that doesn't require confirmation the message prefix should probably be more attestations specific than it is already, at least.

File: packages/docs/operations-manual/key-management.md (hunk 3 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management.md]: Very true. I'm not sure how to easily explain that nuance, which is not terribly relevant for the intended audience, without making things more confusing..

File: packages/docs/operations-manual/key-management.md (hunk 4 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management.md]: I put it here to reinforce which keys you might consider keeping on a ledger, which is part of what these docs are intended to clarify

File: packages/docs/operations-manual/key-management.md (hunk 5 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management.md]: Not strictly, IMO, since you can do this for a validator group as well. But switched it around to be more clear, hopefully

File: packages/docs/operations-manual/key-management/detailed.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
+# Detailed Description of Key Types
+This page provides a detailed description of the various key prototypes found in the Celo protocol, as well as small examples of how to designate key types in the Celo proof-of-stake smart contracts.
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: Detailed Description of Celo Account Roles

File: packages/docs/operations-manual/key-management/detailed.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+## Undesignated
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: Celo Account

Any private key generated for use in the protocol has a corresponding Account and Address. The Address is the last 20 bytes of the hash of the public key which corresponds to private key. These keys...

File: packages/docs/operations-manual/key-management/detailed.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+By default, any private key generated in the Celo protocol does not have a specific role in proof-of-stake. These keys can be used to sign and send transactions that are unrelated to Celo's proof-of-stake protocol, including but by no means limited to transferring Celo Gold, transferring Celo Dollars, exchanging Celo Gold for Celo Dollars, etc.
+
+Undesignated keys can be designated as Locked Gold Accounts or authorized as signer keys on behalf of a Locked Gold Account by sending special transactions using `celocli`.
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: would skip this part till you explain all the different types

File: packages/docs/operations-manual/key-management/detailed.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+
+Note that [ReleaseGold](../../celo-codebase/protocol/release-gold) beneficiary keys are by default Undesignated with respect to proof-of-stake, and that the `ReleaseGold` contract address is what ultimately gets designated as a Locked Gold Account.
+
+
+## Locked Gold Accounts
+
+Locked Gold Account keys have the highest level of privileges in the Celo protocol. These keys can be used to lock and unlock cGLD in order to be used in proof-of-stake. Furthermore, Locked Gold Account keys can be used to authorize other keys to sign transactions and messages on behalf of the Locked Gold Account.
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: I would just use Account everywhere and not use keys since it is implied that you need a private key for an account

File: packages/docs/operations-manual/key-management/detailed.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+# Designate the key as a Locked Gold Account
+celocli account:create --from $ADDRESS_TO_DESIGNATE --useLedger
+
+# Confirm the key was designated as a Locked Gold Account
+celocli account:show $ADDRESS_TO_DESIGNATE
+```
+
+## Authorized Vote Signers
+
+Any Locked Gold Account key may optionally authorize an undesignated key as a vote signing key. Authorized vote signing keys can vote in validator elections and on-chain governance on behalf of the Locked Gold Account.
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: authorize another account to sign votes

File: packages/docs/operations-manual/key-management/detailed.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+# Confirm that the vote signer was authorized
+celocli account:show $LOCKED_GOLD_ACCOUNT
+
+# You can also look up account info via the authorized signer
+celocli account:show $KEY_TO_AUTHORIZE
+```
+
+## Authorized Validator Signers
+
+Any Locked Gold Account key may optionally authorize an undesignated key as a validator signer. Authorized validator signing keys can be used to register and manage a validator or validator group on behalf of the Locked Gold Account. If the authorized validator key is used to register a validator it is also used to sign consensus messages.
```
Comment (cmcewen) [inline packages/docs/operations-manual/key-management/detailed.md]: authorize another account as a Validator Signer

File: packages/docs/operations-manual/key-management/detailed.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
+# Detailed Description of Key Types
+This page provides a detailed description of the various key prototypes found in the Celo protocol, as well as small examples of how to designate key types in the Celo proof-of-stake smart contracts.
+
+## Undesignated
+
+By default, any private key generated in the Celo protocol does not have a specific role in proof-of-stake. These keys can be used to sign and send transactions that are unrelated to Celo's proof-of-stake protocol, including but by no means limited to transferring Celo Gold, transferring Celo Dollars, exchanging Celo Gold for Celo Dollars, etc.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
By default, any private key generated in the Celo protocol does not have a specific role in consensus. Undesignated keys can be used to sign and send transactions that are unrelated to consensus, including but by no means limited to transferring Celo Gold, transferring Celo Dollars, exchanging Celo Gold for Celo Dollars, etc.
```

File: packages/docs/operations-manual/key-management/detailed.md (hunk 8 above)
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
Undesignated keys can be designated as Locked Gold Accounts or authorized as signer keys on behalf of a Locked Gold Account by sending special transactions using [celocli](../../command-line-interface/introduction).
```

File: packages/docs/operations-manual/key-management/detailed.md [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+
+Undesignated keys can be designated as Locked Gold Accounts or authorized as signer keys on behalf of a Locked Gold Account by sending special transactions using `celocli`.
+
+Note that [ReleaseGold](../../celo-codebase/protocol/release-gold) beneficiary keys are by default Undesignated with respect to proof-of-stake, and that the `ReleaseGold` contract address is what ultimately gets designated as a Locked Gold Account.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: May be a bit in the weeds here, might consider moving this into the locked gold section

File: packages/docs/operations-manual/key-management/detailed.md (hunk 9 above)
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
Locked Gold Account keys have the highest level of privilege in the Celo protocol. These keys can be used to lock and unlock cGLD in order to be used in proof-of-stake. Furthermore, Locked Gold Account keys can be used to authorize other keys to sign transactions and messages on behalf of the Locked Gold Account.
```

File: packages/docs/operations-manual/key-management/detailed.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+
+
+## Locked Gold Accounts
+
+Locked Gold Account keys have the highest level of privileges in the Celo protocol. These keys can be used to lock and unlock cGLD in order to be used in proof-of-stake. Furthermore, Locked Gold Account keys can be used to authorize other keys to sign transactions and messages on behalf of the Locked Gold Account.
+
+In *most* cases, the Locked Gold Account key has all the privileges as any authorized keys. For example, if a voter signing key is authorized, a user can place votes on behalf of the Locked Gold Account with both the authorized vote signing key *and* the Locked Gold Account key.
+
+Because of the significant priveleges afforded to the Locked Gold Account key, it is best to store this key securely and access it as infrequently as is possible. Authorizing other signer keys is one way to minimize how frequently you need to access your Locked Gold Account key. The Locked Gold Account key will only be used to send transactions and **can be stored on a Ledger hardware wallet.**
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: awesome 👍

File: packages/docs/operations-manual/key-management/detailed.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+
+### Creating a Locked Gold Account
+An undesignated key may be designated as a Locked Gold Account key by running the following command:
+
+```shell
+# Designate the key as a Locked Gold Account
+celocli account:create --from $ADDRESS_TO_DESIGNATE --useLedger
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md:23]: are addresses public key hashes?  or are key / address interchangeable?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+
+# Confirm the key was designated as a Locked Gold Account
+celocli account:show $ADDRESS_TO_DESIGNATE
+```
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md:27]: TODO [later]: explain after this that the accounts module in celocli deals with all Locked Gold Accounts

File: packages/docs/operations-manual/key-management/detailed.md (hunk 13 above)
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
Note that [ReleaseGold](../../celo-codebase/protocol/release-gold) beneficiary keys are by default undesignated with respect to proof-of-stake, and that the `ReleaseGold` contract address is what ultimately gets designated as a Locked Gold Account.
```

File: packages/docs/operations-manual/key-management/detailed.md (hunk 10 above)
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
Any Locked Gold Account key may optionally authorize an undesignated key as a vote signing key. Authorized vote signing keys can vote for validator groups and for on-chain governance proposals on behalf of the Locked Gold Account.
```

Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: do we call them `vote signer keys` or `vote signing keys`? should change the header depending on which.. I vote for vote signer

File: packages/docs/operations-manual/key-management/detailed.md [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+
+## Authorized Vote Signers
+
+Any Locked Gold Account key may optionally authorize an undesignated key as a vote signing key. Authorized vote signing keys can vote in validator elections and on-chain governance on behalf of the Locked Gold Account.
+
+Note that the vote signing key must first be used to generate a "proof-of-possession" indicating that key's willingness to be authorized as a signer on behalf of the Locked Gold Account.
+
+Authorized vote signing keys will only be used to send transactions and **can be stored on a Ledger hardware wallet**.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: worth specifying here that they cannot be used to move funds?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 18]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+## Authorized Validator Signers
+
+Any Locked Gold Account key may optionally authorize an undesignated key as a validator signer. Authorized validator signing keys can be used to register and manage a validator or validator group on behalf of the Locked Gold Account. If the authorized validator key is used to register a validator it is also used to sign consensus messages.
+
+### Authorized Validator Signers for Validator Groups
+An authorized validator signer key that will be used to register a validator group can be used to send group management transactions (e.g. register, add member A, queue commission update to 0.25, etc.) Because this key does not participate directly in consensus it **can be stored on a Ledger hardware wallet.**
+
+### Authorized Validator Signers for Validators
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md:64]: naive reader hat -- can I use the same validator signer for both group and validator?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 19]
```diff
@@ -0,0 +1,148 @@
...
+An authorized validator signer key that will be used to register a validator can be used to send validator management transactions (e.g. register, affiliate with group A, etc.) This key will also be used to sign consensus messages and thus *cannot be stored on a Ledger hardware wallet* as signing consensus messages is not currently supported by the Celo Ledger App.
+
+Note that the validator signing key must first be used to generate a "proof-of-possession" indicating that key's willingness to be authorized as a signer on behalf of the Locked Gold Account.
+
+
+### Authorizing a Validator Signer
+An undesignated key may be authorized as a validator signer on behalf of a Locked Gold Account by running the following commands:
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: note should we use `LockedGold` to refer to the contract and add a link somewhere? It is referenced a lot

File: packages/docs/operations-manual/key-management/detailed.md [hunk 20]
```diff
@@ -0,0 +1,148 @@
...
+# Confirm that the vote signer was authorized
+celocli account:show $LOCKED_GOLD_ACCOUNT
+
+# You can also look up account info via the authorized signer
+celocli account:show $KEY_TO_AUTHORIZE
+```
+
+## Authorized Validator BLS Signers
+
+The Celo protocol uses BLS signatures in consensus to ultimately determine whether or not a particular block is valid. Many BLS signatures over the same content can be combined into a single "aggregated signature", allowing several kilobytes of signatures to be compressed into fewer than 100 bytes, ensuring that the block headers remain compact and mobile friendly.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
The Celo protocol uses BLS signatures in consensus to ultimately determine whether or not a particular block is valid. Many BLS signatures over the same content can be combined into a single "aggregated signature", allowing several kilobytes of signatures to be compressed into fewer than 100 bytes, ensuring that the block headers remain compact and light client friendly.
```

File: packages/docs/operations-manual/key-management/detailed.md [hunk 21]
```diff
@@ -0,0 +1,148 @@
...
+
+When registering a Validator on behalf of a Locked Gold Account, users must provide a BLS public key, as well as a proof-of-possession to protect against [rogue key attacks](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html).
+
+By default users can derive the BLS key directly from their authorized validator signer key. From a key management perspective, this means that the authorized BLS signer key is **exactly the same** as the authorized validator signer key.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: security perspective?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 22]
```diff
@@ -0,0 +1,148 @@
...
+
+Most users will only need to think about BLS signer keys when registering a validator, or when authorizing a new validator signing key *after* registering a validator. It follows that when a validator authorizes a new validator signer key, the BLS public key and proof-of-possession for the new authorized validator signer should be provided as well.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: what happens if I keep using the old BLS key?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 23]
```diff
@@ -0,0 +1,148 @@
...
+celocli validator:register --from $AUTHORIZED_VALIDATOR_SIGNER --blsKey $BLS_PUBLIC_KEY --blsSignature $BLS_PROOF_OF_POSSESSION
+
+# Confirm that the validator was registered
+celocli validator:show $LOCKED_GOLD_ACCOUNT
+
+# You can also look up the validator via the authorized signer
+celocli validator:show $AUTHORIZED_VALIDATOR_SIGNER
+```
+
+## Authorized Attestations Signers
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]: Attestation plural?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 24]
```diff
@@ -0,0 +1,148 @@
...
+
+## Authorized Validator Signers
+
+Any Locked Gold Account key may optionally authorize an undesignated key as a validator signer. Authorized validator signing keys can be used to register and manage a validator or validator group on behalf of the Locked Gold Account. If the authorized validator key is used to register a validator it is also used to sign consensus messages.
+
+### Authorized Validator Signers for Validator Groups
+An authorized validator signer key that will be used to register a validator group can be used to send group management transactions (e.g. register, add member A, queue commission update to 0.25, etc.) Because this key does not participate directly in consensus it **can be stored on a Ledger hardware wallet.**
+
+### Authorized Validator Signers for Validators
+An authorized validator signer key that will be used to register a validator can be used to send validator management transactions (e.g. register, affiliate with group A, etc.) This key will also be used to sign consensus messages and thus *cannot be stored on a Ledger hardware wallet* as signing consensus messages is not currently supported by the Celo Ledger App.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/detailed.md]:
```suggestion
An authorized validator signer key that will be used to register a validator can be used to send validator management transactions (e.g. register, affiliate with group A, etc.) This key will also be used to sign consensus messages and thus **cannot be stored on a Ledger hardware wallet** as signing consensus messages is not currently supported by the Celo Ledger App.
```

File: packages/docs/operations-manual/key-management/key-rotation.md [hunk 25]
```diff
@@ -0,0 +1,40 @@
+## Key Rotation
+
+If a signing key is lost or compromised, the Locked Gold Account key can authorize a new signing key to replace the old one. This prevents losing a signing key from becoming a catastrophic event and it is strongly recommended to regularly rotate keys to limit the impact of an undiscovered compromise.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/key-rotation.md]: any general heuristics on schedule for key rotation? what is the tradeoff in having to use colder account keys?

File: packages/docs/operations-manual/key-management/summary.md [hunk 26]
```diff
@@ -0,0 +1,33 @@
+# Key management
+
+> Crypto is a tool for turning a whole swathe of problems into key management problems. Key management problems are way harder than (virtually all) cryptographers think.
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/summary.md:3]: amazing

File: packages/docs/operations-manual/key-management/summary.md [hunk 27]
```diff
@@ -0,0 +1,33 @@
...
+>
+> [@LeaKissner on Twitter](https://twitter.com/LeaKissner/status/1198595109756887040)
+
+
+## Philosophy
+
+The Celo protocol was designed with the understanding that there is often an inherent tradeoff between the convenience of accessessing a private key and the security with which that private key can be custodied. As such, the protocol allows users to authorize private keys with specific, limited privileges. This allows users to custody each private key according to its sensitivity (i.e. what is the impact of this key being lost or stolen?) and usage patterns (i.e. how often and under which circumstances will this key need to be accessed).
```
Comment (pranaymohan) [inline packages/docs/operations-manual/key-management/summary.md]:
```suggestion
The Celo protocol was designed with the understanding that there is often an inherent tradeoff between the convenience of accessing a private key and the security with which that private key can be custodied. In general Celo is unopinionated about how keys are custodied, but also allows users to authorize private keys with specific, limited privileges. This allows users to custody each private key according to its sensitivity (i.e. what is the impact of this key being lost or stolen?) and usage patterns (i.e. how often and under which circumstances will this key need to be accessed).
```

**Review by pranaymohan — COMMENTED** [review]
This will be incredibly helpful. Thanks for adding this

File: packages/docs/operations-manual/key-management/detailed.md (hunk 9 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Yeah I think your suggestion around "celo accounts" lead towards a much clearer and consistent terminology. Only using "key" from here on out when referencing the private key material directly (e.g. keep your key on a ledger)

File: packages/docs/operations-manual/key-management/detailed.md (hunk 10 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Calling these authorized vote signers everywhere now, I think it's much more consistent

File: packages/docs/operations-manual/key-management/detailed.md (hunk 12 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Outdated after @cmcewen's feedback

File: packages/docs/operations-manual/key-management/detailed.md (hunk 8 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Accepting manually

File: packages/docs/operations-manual/key-management/detailed.md (hunk 9 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Accepted manually as already editing

File: packages/docs/operations-manual/key-management/detailed.md (hunk 15 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md:23]: Used inconsistently here, tried to overhaul

File: packages/docs/operations-manual/key-management/detailed.md (hunk 13 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Removed this terminology

File: packages/docs/operations-manual/key-management/detailed.md (hunk 10 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: +1

Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: +1, made consistent everywhere

File: packages/docs/operations-manual/key-management/detailed.md (hunk 17 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: "can only be used to send voting txs"

File: packages/docs/operations-manual/key-management/detailed.md (hunk 19 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Linked to the docs page on locked gold

File: packages/docs/operations-manual/key-management/detailed.md (hunk 20 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Done

File: packages/docs/operations-manual/key-management/detailed.md (hunk 21 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: "key management and security perspective"?

I think folks will want to realize that this isn't exactly another key that they need to custody

File: packages/docs/operations-manual/key-management/detailed.md (hunk 22 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Out of scope for this documentation. It would work fine so long as you still had the BLS key accessible

File: packages/docs/operations-manual/key-management/detailed.md (hunk 23 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/detailed.md]: Done

File: packages/docs/operations-manual/key-management/key-rotation.md (hunk 25 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/key-rotation.md]: Not really, it depends on the operational complexity, security of account keys, etc. Maybe on the order of monthly from a security standpoint, sometime when you would need to access your account key anyway for something else

File: packages/docs/operations-manual/key-management/summary.md (hunk 26 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/summary.md:3]: h/t @nambrot for finding this tweet

File: packages/docs/operations-manual/key-management/summary.md (hunk 27 above)
Comment (asaj) [reply packages/docs/operations-manual/key-management/summary.md]: Accepted manually

File: packages/docs/operations-manual/key-management/detailed.md [hunk 28]
```diff
@@ -0,0 +1,146 @@
+# Detailed Description of Celo Account Roles
+This page provides a detailed description of the various account rolese found in the Celo protocol, as well as small examples of how to designate and account as playing a particular role.
```
Comment (clrblmt) [inline packages/docs/operations-manual/key-management/detailed.md:2]: Typo "rolese" - should be $roles"

File: packages/docs/operations-manual/key-management/detailed.md [hunk 29]
```diff
@@ -0,0 +1,146 @@
...
+
+## Celo Accounts
+
+Any private key generated for use in the Celo protocol has a corresponding address. The account address is the last 20 bytes of the hash of the corresponding public key, just as in Ethereum. Celo account keys can be used to sign and send transactions on the Celo network.
+
+Celo Accounts can be designated as Locked Gold Accounts or authorized as signer keys on behalf of a Locked Gold Account by sending special transactions using [celocli](../../command-line-interface/introduction). Note that Celo accounts that have not been designated as Locked Gold Accounts or authorized signers may not be able to send certain transactions related to proof-of-stake.
+
+## Locked Gold Accounts
+
+[Locked Gold](../../celo-codebase/protocol/proof-of-stake/locked-gold.md) Account keys have the highest level of privilege in the Celo protocol. These keys can be used to lock and unlock cGLD in order to be used in proof-of-stake. Furthermore, Locked Gold Account keys can be used to authorize other keys to sign transactions and messages on behalf of the Locked Gold Account.
```
Comment (clrblmt) [inline packages/docs/operations-manual/key-management/detailed.md:12]: is cGLD defined anywhere?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 30]
```diff
@@ -0,0 +1,146 @@
...
+
+In *most* cases, the Locked Gold Account key has all the privileges as any authorized signers. For example, if a voter signer is authorized, a user can place votes on behalf of the Locked Gold Account with both the authorized vote signer *and* the Locked Gold Account.
+
+Because of the significant priveleges afforded to the Locked Gold Account, it is best to store this key securely and access it as infrequently as is possible. Authorizing other signers is one way to minimize how frequently you need to access your Locked Gold Account key. The Locked Gold Account key will only be used to send transactions and **can be stored on a Ledger hardware wallet.**
```
Comment (clrblmt) [inline packages/docs/operations-manual/key-management/detailed.md:16]: The Locked Gold Account key MAY only be used to 

Link to Ledger hardware wallet instructions?

File: packages/docs/operations-manual/key-management/detailed.md [hunk 31]
```diff
@@ -0,0 +1,146 @@
...
+
+```shell
+# Designate the Celo account as a Locked Gold Account
+celocli account:create --from $ADDRESS_TO_DESIGNATE --useLedger
+
+# Confirm the address was designated as a Locked Gold Account
+celocli account:show $ADDRESS_TO_DESIGNATE
+```
+
+Note that [ReleaseGold](../../celo-codebase/protocol/release-gold) beneficiary keys are considered vanilla Celo accounts with respect to proof-of-stake, and that the `ReleaseGold` contract address is what ultimately gets designated as a Locked Gold Account.
```
Comment (clrblmt) [inline packages/docs/operations-manual/key-management/detailed.md:29]: "vanilla" may not make sense for a non-native English speaker

File: packages/docs/operations-manual/key-management/key-rotation.md [hunk 32]
```diff
@@ -0,0 +1,40 @@
...
+
+  # On a node containing your Locked Gold Account key.
+  celocli account:authorize --from $LOCKED_GOLD_ACCOUNT --role validator --signer $SIGNER_TO_AUTHORIZE --signature $SIGNER_PROOF_OF_POSSESSION --blsKey $BLS_PUBLIC_KEY --blsPop $BLS_PROOF_OF_POSSESSION
+  ```
+
+3. **Leave all validator and proxy nodes running** until the next epoch change. At the start the next epoch, the new Validator signer should take over participation in consensus.
+
+4. Verify that key rotation was successful. Here are some ways to check:
+  <!-- TODO: The following URL assumes that the user is running against the baklava network. This will need to be updated -->
+  * Open `baklava-blockscout.celo-testnet.org/address/<SIGNER_TO_AUTHORIZE>/validations` to confirm that blocks are being proposed.
```
Comment (clrblmt) [inline packages/docs/operations-manual/key-management/key-rotation.md:32]: When mainnet launched: Thoughts on how to communicate this info for Baklava testnet w/out too much redundancy?

File: packages/docs/operations-manual/key-management/summary.md (hunk 26 above)
Comment (clrblmt) [reply packages/docs/operations-manual/key-management/summary.md:3]: +1 love it

**Review by clrblmt — COMMENTED** [review]
Minor edits / comments
